packages/formspree-core/src/core.ts

```diff
@@ -46,22 +43,17 @@ export class Client {
    * @param data - An object or FormData instance containing submission data.
    * @param args - An object of form submission data.
    */
-  async submitForm(
+  async submitForm<T extends FieldValues>(
```

Comment:
Get type-safety based on the `SubmissionData`, the error fields also uses this type information. ✨

Inspired by react-hook-form. 🙏

Reply:
Users don't have to pass type parameter; it's inferred by the shape of the `SubmissionData`. For example, if a user call `.submitForm('form-key', { email: 'name@example.com', message: '123' })`, `T` will be inferred as `{email: string; message: string}`. Moreover, when user of the lib calls `errResult.getFieldErrors(...)`, the field names are typed to `keyof T` so they get autocomplete + type safety (prevent accidentally try to grab the wrong field).